src/horus_demod.c

```diff
@@ -65,10 +68,12 @@ int main(int argc, char *argv[]) {
             {"help",      no_argument,        0, 'h'},
             {"mode",      required_argument,  0, 'm'},
             {"stats",     optional_argument,  0, 't'},
+            {"fsk_lower", optional_argument,  0, 'b'},
+            {"fsk_upper", optional_argument,  0, 'u'},
             {0, 0, 0, 0}
```

Comment:
Should this be required_argument?  The you won't need the ```if (optarg != NULL){``` below.

Reply:
fsk_lower and fsk_upper are both optional arguments, and if present (detected by fsk_lower >0) override whatever the fsk demod sets up on initialisation.
Have I done the argument parsing correctly in this case?

Reply:
Yes you are right - I think what we need is the ```b:u:``` in the comment below, to force the user to enter the numerical arguments for the ```fsk_lower``` and ```fsk_upper ``` options.